eo-maven-plugin/src/main/java/org/eolang/maven/DependenciesFile.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2016-2022 Objectionary.com
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.eolang.maven;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import org.apache.maven.model.Dependency;
+
+/**
+ * File with the list of dependencies.
+ *
+ * @since 0.28.11
+ */
+interface DependenciesFile {
+
+    /**
+     * Returns the link to the file with the list of dependencies.
+     *
+     * @param origin Dependency which transitive dependencies we are interested of
+     * @return File with dependencies
+     */
+    Path file(Dependency origin);
```

Comment:
@volodya-lombrozo It is still awkward design here. We have a **File** object which has _file_ attribute. I would expect a **File** to have _path_. 
Also this method accepts particular dependency - does that mean that returned path would contain a file with dependencies for this dependency only?
And `DependenciesFile` represents a file with all possible transitive dependencies?
How would I see it:
```
Path path = new DependenciesFileImpl(origin).path();
```

Reply:
@mximp I've implemented the new structure of classes. Please check. Some points:
1) We don't have `TransitiveDependencies` anymore.
2) Also `DependenciesFile` and connected implementations were removed.
3) Predicates still exists, because if we will define them right into `ResolveMojo` (as lambdas) - we won't be able to test them. Also it's impossible to define them: 
3.1)  as `static field`  - we have to provide particular dependency for `NotSame` predicate. 
3.2) as `factory method` - pure design
3.3) Of course, we can introduce one more specific class, like `DcsTransitive`:
```java
public class DcsTransitive implements Dependencies {

    private final Dependencies delegate;

    public DcsTransitive(final Dependencies delegate, final Dependency origin) {
        this.delegate = new DcsFiltered(
            delegate,
            Arrays.asList(
                new DcsFiltered.NotSame(origin),
                new DcsFiltered.NotRuntime(),
                new DcsFiltered.NotTesting()
            )
        );
    }

    @Override
    public Collection<Dependency> all() {
        return delegate.all();
    }
}
```
but, as I understand, you against it, because it's an overdesign.

Reply:
@volodya-lombrozo Actually, it's a very good idea... `DcsTransitive` in this form looks fine in this form. Can you add it and use?

Reply:
@mximp Added.